src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2835,6 +2837,53 @@ public override IOperation VisitCoalesce(ICoalesceOperation operation, int? capt
             return result;
         }
 
+        public override IOperation VisitCoalesceAssignment(ICoalesceAssignmentOperation operation, int? captureIdForResult)
+        {
+            SpillEvalStack();
+
+            RegionBuilder resultCaptureRegion = _currentRegion;
+
+            EvalStackFrame frame = PushStackFrame();
+
+            var whenNull = new BasicBlockBuilder(BasicBlockKind.Block);
+            IOperation testExpression = NullCheckAndConvertCoalesceValue(operation.Target, coalesceConversion: null, operation.Type, whenNull);
+
+            // Coalesce can have conversions on the left side, but coalesce assignment should not.
+            Debug.Assert(testExpression.Kind != OperationKind.Conversion);
+
+            var afterCoalesce = new BasicBlockBuilder(BasicBlockKind.Block);
+
+            int resultCaptureId = captureIdForResult ?? GetNextCaptureId(resultCaptureRegion);
+
+            AddStatement(new FlowCapture(resultCaptureId, operation.Target.Syntax, testExpression));
+
+            LinkBlocks(CurrentBasicBlock, afterCoalesce);
+            _currentBasicBlock = null;
+
+            AppendNewBlock(whenNull);
+
+            // The return of Visit(operation.WhenNull) can be a flow capture that wasn't used in the non-null branch. We want to create a
+            // region around it to ensure that the scope of the flow capture is as narrow as possible. If there was no flow capture, region
+            // packing will take care of removing the empty region.
+            var whenNullRegion = new RegionBuilder(ControlFlowRegionKind.LocalLifetime, isStackSpillRegion: true);
```

Comment:
>var whenNullRegion = new RegionBuilder(ControlFlowRegionKind.LocalLifetime, isStackSpillRegion: true); [](start = 12, length = 102)

Does this really help? It feels like ```PopStackFrame(frame);``` below can still merge this region into its parent. It might be simpler and more robust to allocate a new frame and use PopStackFrameAndLeaveRegion for it after ```AddStatement(new FlowCapture(resultCaptureId, operation.Syntax, nullAssignment));```